utils/commands/doc.py

```diff
@@ -215,6 +215,39 @@ def _run(self, language, target_dir, browser, skip_screenshots=True, force=False
         with open(os.path.join(target_dir, "..", "version"), "w+") as f:
             f.write(self._get_source_version())
 
+        # create symlinks
+        symlinkname = ''
+        git = sh.Command("git")
+        branch = git("rev-parse", "--abbrev-ref", "HEAD").strip() if os.environ.get('TRAVIS_BRANCH') is None \
+            else os.environ.get('TRAVIS_BRANCH')
+
+        if branch == "develop":
+            # handle develop builds:
+            print('detected development build')
+            symlinkname = self.config.get("DEFAULT", "develop-version-mapping")
+        else:
+            # handle releases:
+            # TODO: this is assuming we are always building the documentation of
+            # the most recent version - which is usually true.
+            # When for some strange reasons an older version is being built but
+            # a newer is also available it will overwrite the symlink to the
+            # newer version.
+            # Simple work around: just retrigger a build of the newer version
+            # afterwards manually.
+            print('detected build of most recent version')
+            symlinkname = self.config.get("DEFAULT", "most-recent-version-mapping")
+
+        if '' != symlinkname:
+            print("setting symlink '%s' to '%s'" % (symlinkname, target_dir))
+            cwd = os.getcwd()
+            os.chdir(root)
```

Comment:
This line throws an error when i generate the documentation:
`NameError: global name 'root' is not defined`

Reply:
Fixed in 9d65b21